src/manager.js

```diff
@@ -305,50 +320,99 @@ AdManager.prototype.setSlotTargeting = function(element, slot) {
  * @returns {Element} - Fully configured ad slot
 */
 AdManager.prototype.configureAd = function (element) {
+
+  var slotName = this.options.dfpSiteCode || "",
+    kinjaMeta = window.kinja.meta,
+    postMeta = window.kinja.postMeta || {},
+    currentBlog = kinjaMeta.blog,
+    blogSales = kinjaMeta.blogSales,
+    pageType = kinjaMeta.pageType,
+    getExperimentVariation = function(scope) {
+      var expScope = scope || window,
+        variation = expScope.cxApi ? expScope.cxApi.getChosenVariation(expScope.gaExperimentId) : null;
+      return (variation !== null && variation >= 0 && variation < 26) ? String.fromCharCode(variation + 65) : null;
+    },
+
+    getExperimentId = function(scope) {
+      var expScope = scope || window;
+      return (expScope.gaExperimentId !== undefined) ? expScope.gaExperimentId : null;
+    },
+
+    getQueryParameter = function(name) {
+      var regexS = '[\\?&]' + name + '=([^&#]*)',
+        regex = new RegExp(regexS),
+        results = regex.exec(window.location.search),
+        retval = '';
+
+      if (results) {
+        return decodeURIComponent(results[1].replace(/\+/g, ' '));
+      }
+
+      return retval;
+    },
+    expVar = (getExperimentVariation() !== null && getExperimentId() !== null) ? getExperimentId() + '_' + getExperimentVariation() : null,
+    paramZone = getQueryParameter('adzone'),
+    forcedAdZone = (function(pZone, meta) {
```

Comment:
We should also break this into a separate, named function that so we can unit test the functionality within it.